core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -455,7 +455,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     _ui =
       if (conf.getBoolean("spark.ui.enabled", true)) {
         Some(SparkUI.createLiveUI(this, _conf, listenerBus, _jobProgressListener,
-          _env.securityManager, appName, startTime = startTime))
+          _env.securityManager, applicationId, appName, startTime = startTime))
```

Comment:
`applicationId` hasn't been initialized yet here, so you shouldn't need to provide it in the constructor. Just call `setAppId` later as you're already doing.